Artroom1/ArtViewController.swift

```diff
@@ -34,20 +34,50 @@ class ArtViewController: UIViewController {
        
      }
     
-    @IBAction func renderFresh() {
+     func renderFresh() {
         
         if selection.isEmpty {
             self.artResults.text = "No more art matches"
             self.artImage.image = UIImage (named: "Empty Frame")
-            anotherButton.isEnabled = false
-            anotherButton.alpha = 0.5
+//            anotherButton.isEnabled = false
+//            anotherButton.alpha = 0.5
 
         } else {
             renderNextArt()
         }
        
     }
     
+    @IBAction func handlePan(recognizer:UIPanGestureRecognizer) {
+        let translation = recognizer.translation(in: self.view)
+        var thisIsATestArtArray: [Artwork] = []
+        if let view = recognizer.view {
+            view.center = CGPoint(x:view.center.x + translation.x,
+                                  y:view.center.y + translation.y)
+        }
+        recognizer.setTranslation(CGPoint.zero, in: self.view)
+        
+        
+        if recognizer.state == UIGestureRecognizer.State.ended {
+            
+            let velocity = recognizer.velocity(in: self.view)
+            if(velocity.x > 0) {
```

Comment:
I am not sure if what I am saying is right, but I guess that if you drag an image close to the left (or right) corner of the phone and then move slightly back to the center, it will detect it as a swipe right, although you are on the left side of the screen. I don't know if that is the behavior you want.
Maybe it would be a nice thing to implement this _snap back_ behavior we talked about after the standup. So only count it as a swipe left or right if at least a certain distance to the center has been reached and if the velocity is not in the opposite direction.

Reply:
We can also pair on that on Monday if you want.